### PR TITLE
[ECW-2933] Update validation regex across a set of tokens

### PR DESCRIPTION
**Pull request by iurevych**
This PR updates incorrect validation regexes for a series of tokens. This was a [result of a manual QA](https://docs.google.com/spreadsheets/d/1F7bK-KVqb7TuQTQu0HAD-dF55HTTE5-S4DkAWJ8vs2c/edit?usp=sharing) on our end.

### 1. Contracts versioning
- [ ] Make sure that the `patch` version of the contracts is increased if changes have been made to the `UNSRegistry`, `MintingManager`, `ProxyReader`, `ENSCustody` contracts.
- [ ] Make sure that the `minor` version of the contracts is increased if breaking changes have been made to the `UNSRegistry`, `MintingManager`, `ProxyReader`, `ENSCustody` contracts. It includes changes of interfaces.
### 2. Contracts licensing
- [ ] Make sure that no **SPDX-License-Identifier** defined in contracts.
- [ ] Make sure that the **header** is added to the new contract files.
  ```
  // @author Unstoppable Domains, Inc.
  // @date {Month} {Day}(ordinal), {Year}
  ```
### 3. Coverage
- [ ] Make sure that the coverage of contracts has not decreased and strive **100%**
### 4. Configs versioning
- [ ] Make sure that the version of `uns-config.json` is increased if changes have been made to the config.
- [ ] Make sure that the version of `ens-config.json` is increased if changes have been made to the config.
- [x] Make sure that the version of `resolver-keys.json` is increased if changes have been made to the config.
- [x] Make sure that the version of `ens-resolver-keys.json` is increased if changes have been made to the config.
### 5. Package versioning
- [x] Make sure that the `patch` version of package is increased if valuable changes have been made to the package. It includes contracts update, configs update, etc.
- [ ] Make sure that the `major.minor` version of package is synced with version of `UNSRegistry` contract.
- [ ] Make sure that the `CHANGELOG` is updated with short description for the new version.
### 6. Code review
- [x] `resolver-keys.json` code review is required from **DevTools** team
- [x] `ens-resolver-keys.json` code review is required from **DevTools** team
- [ ] For all other changes code review is required from **Registry** team

### CODE REVIEW

File: ens-resolver-keys.json [hunk 1]
```diff
@@ -594,7 +594,7 @@
       "unsKey": "crypto.STX.address",
       "symbol": "STX",
       "label": "Stacks",
-      "validationRegex": "^[a-zA-Z0-9]*$"
+      "validationRegex": "SP[A-Z0-9]{38,39}$"
```
Comment (DeRain) [inline ens-resolver-keys.json:597]:
```suggestion
      "validationRegex": "^SP[A-Z0-9]{38,39}$"
```

File: ens-resolver-keys.json [hunk 2]
```diff
@@ -142,7 +142,7 @@
       "unsKey": "crypto.EGLD.address",
       "symbol": "EGLD",
       "label": "MultiversX",
-      "validationRegex": "^0x[a-zA-Z0-9]*$"
+      "validationRegex": "erd[a-zA-Z0-9]{59}$"
```
Comment (DeRain) [inline ens-resolver-keys.json:145]:
```suggestion
      "validationRegex": "^erd[a-zA-Z0-9]{59}$"
```

File: resolver-keys.json [hunk 3]
```diff
@@ -1423,7 +1423,7 @@
     },
     "crypto.EGLD.address": {
       "deprecatedKeyName": "EGLD",
-      "validationRegex": "^0x[a-zA-Z0-9]*$",
+      "validationRegex": "erd[a-zA-Z0-9]{59}$",
```
Comment (DeRain) [inline resolver-keys.json:1426]:
```suggestion
      "validationRegex": "^erd[a-zA-Z0-9]{59}$",
```

File: resolver-keys.json [hunk 4]
```diff
@@ -1528,7 +1528,7 @@
     },
     "crypto.STX.address": {
       "deprecatedKeyName": "STX",
-      "validationRegex": "^[a-zA-Z0-9]*$",
+      "validationRegex": "SP[A-Z0-9]{38,39}$",
```
Comment (DeRain) [inline resolver-keys.json:1531]:
```suggestion
      "validationRegex": "^SP[A-Z0-9]{38,39}$",
```

**Review by DeRain — APPROVED** [review]

**Review by sammyluo — COMMENTED** [review]
Looks like some of the regex changes are breaking changes. I just checked resolution service and libs and they should be fine since they don't use the regex filed. Do you know if it will break the existing records on other products?

**Comment by iurevych** [comment]
@sammyluo I'm only aware of the website and the mobile app which rely on these configs. Even if the existing records are already saved, those records are 💯 invalid.

**Review by osa911 — APPROVED** [review]

**Pull request merged**